query-engine/prisma-models/src/internal_data_model.rs

```diff
@@ -110,6 +109,14 @@ impl InternalDataModel {
             })
             .as_slice()
     }
+
+    pub fn walk<I>(&self, id: I) -> psl::parser_database::walkers::Walker<I> {
+        self.schema.db.walk(id)
+    }
+
+    pub fn zip<I>(self: Arc<InternalDataModel>, id: I) -> crate::Zipper<I> {
+        crate::Zipper { id, dm: self }
+    }
```

Comment:
Is the goal of making `self` an `Arc<T>` to have the caller explicitly clone the datamodel? Didn't even know you can specify a type for `self` 🙈

Reply:
yes we could take `&self` here and clone inside the `zip()` function, but that would _always_ clone. By forcing the  caller to give you an owned Arc, you give the opportunity to the caller to avoid cloning if it can.

Reply:
I follow https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control religiously.

Reply:
Love it, thanks for the explanation 💯